### ISSUE TITLE
Bun crashed when installing packages with `bun i` on Windows 10

### ISSUE THREAD

**Issue by azivkovi** [oven-sh/bun#10845]
### How can we reproduce the crash?

I ran npm-check-updates which resulted in the following updates for package.json:
```md
 @sentry/cli                    ^2.31.1  →    ^2.31.2
 aws-sdk                      ^2.1611.0  →  ^2.1613.0
 eslint                          8.57.0  →      9.2.0
 react-intersection-observer    ^9.10.1  →    ^9.10.2
 ```

I then ran `bun i`

### Relevant log output

```shell
Bun v1.1.7 (b0b7db5c) Windows x64
Args: "C:\Users\azivkovi\.bun\bin\bun.exe", "i"
Features: dotenv extracted_packages(21)
Elapsed: 28115ms | User: 1531ms | Sys: 5765ms
RSS: 0.39GB | Peak: 0.43GB | Commit: 0.44GB | Faults: 181828

panic(main thread): index out of bounds
oh no: Bun has crashed. This indicates a bug in Bun, not your code.
```

### Stack Trace (bun.report)

Bun v1.1.7 ([`b0b7db5`](https://github.com/oven-sh/bun/tree/b0b7db5c060e830801b0273b7ff6e34e3eb5c254)) on windows x86_64 [InstallCommand]

**panic**: index out of bounds

- [`install.zig:4092`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/install/install.zig#L4092): `enqueueDependencyWithMain`
- [`install.zig:4932`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/install/install.zig#L4932): `processDependencyList`
- [`install.zig:5652`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/install/install.zig#L5652): `runTasks`
- [`install.zig:10135`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/install/install.zig#L10135): `installPackages`
- [`install.zig:10815`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/install/install.zig#L10815): `installWithManager`
- [`filter_run.zig:428`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/cli/filter_run.zig#L428): `install`
- [`cli.zig:1440`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/cli.zig#L1440): `start`
- [`cli.zig:62`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/cli.zig#L62): `start`
- [`main.zig:51`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/main.zig#L51): `main`
- [`exe_common.inl:78`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/src/startup/exe_common.inl#L78): `invoke_main`

**Comment by azivkovi** [comment]
`npm i` results in the following error, posting in case it is relevant:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: primo@1.0.0
npm ERR! Found: eslint@9.2.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"9.2.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^7.23.0 || ^8.0.0" from eslint-config-next@14.2.3
npm ERR! node_modules/eslint-config-next
npm ERR!   dev eslint-config-next@"14.2.3" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by Jarred-Sumner** [comment]
Duplicate of https://github.com/oven-sh/bun/issues/9808

**Comment by Jarred-Sumner** [comment]
Closing as duplicate, we will fix this but please follow along in #9808
